src/Travers.php

```diff
@@ -65,6 +88,40 @@ public function find($key)
         return $this->parseParam($this->parseKeys($key), $this->tree);
     }
 
+    /**
+     * @param       $key
+     * @param array $tree
+     *
+     * @return array|mixed
+     *
+     * @throws BranchNotFoundException
+     */
+    protected function removeParam($key)
+    {
+        $result    = $this->tree;
+        $tmp       = &$result;
+        $keysArray = $this->parseKeys($key);
+
+        while (count($keysArray) > 1) {
```

Comment:
`while ($keys) {`

Reply:
I need to stop while when the array has 2 items.